### PR TITLE
Prevent API calls when there are no edit company form changes

### PR DESCRIPTION
**Pull request by paulgain**
## Description of change

Prevent users from submitting the Edit Company Form when they haven't made a change.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

**Review by cgsunkel — APPROVED** [review]

**Comment by adamledwards** [comment]
bit of a UI question should the is the form is pristine should there submit button be disabled? 
Or should we redirect to the success page without calling sending data to the api.
Seem a bit broken at the moment

**Review by PippoRaimondi — APPROVED** [review]

**Review by ambrozic — CHANGES_REQUESTED** [review]

**Comment by paulgain** [comment]
> bit of a UI question should the is the form is pristine should there submit button be disabled?
> Or should we redirect to the success page without calling sending data to the api.
> Seem a bit broken at the moment

I think the GDS guidelines is to not disable buttons (i think). I don't think we do that in Data Hub anyway do we?

Previously I added a blue banner with a message to indicate nothing was saved, something like "No changes detected!". However, Raf was not a fan so I took it out. It's more of a UX question I suppose.

I like the idea of redirecting to the details page though without any success message.

File: src/apps/companies/apps/edit-company/client/EditCompanyForm.jsx [hunk 1]
```diff
@@ -25,7 +25,12 @@ function EditCompanyForm({
   oneListEmail,
   isOnOneList,
 }) {
-  async function onSubmit(values) {
+  async function onSubmit(values, isPrestine) {
```
[the remark this answers is not in the document]
Comment (paulgain) [reply src/apps/companies/apps/edit-company/client/EditCompanyForm.jsx:28]: Can you please elaborate. Thank you.

[the remark this answers is not in the document]
Comment (ambrozic) [reply src/apps/companies/apps/edit-company/client/EditCompanyForm.jsx:28]: `prestine` is grammatically incorrect. it should be `pristine`.

[the remark this answers is not in the document]
Comment (paulgain) [reply src/apps/companies/apps/edit-company/client/EditCompanyForm.jsx:28]: Good spot, thank you 👍

**Comment by ambrozic** [comment]
Hiding submit button when there is no change yet in the form is good or bad practice?

**Review by ambrozic — APPROVED** [review]

**Review by adamledwards — APPROVED** [review]

**Comment by paulgain** [comment]
> Hiding submit button when there is no change yet in the form is good or bad practice?

I don't think we hide buttons per se in Data Hub, however, we do hide data hidden in tabs, accordions and details but these are different. 

I don't think GDS recommend hiding elements on the page as it confuses users although I can't find a link to back up my claims.

**Comment by paulgain** [comment]
@adamledwards @ambrozic @rafenden 

user-options

**These are the options when a user doesn’t modify the form:**
1. User clicks `Submit`, we stay on the same page and do nothing (Adam rightly mentioned maybe perceived as broken)
2. User clicks `Submit`, we stay on the same page and show a blue display banner along with a message (Raf doesn't like this)
3. User clicks `Submit`, we redirect to the Business Details page without showing the green success banner (same as clicking `Return without saving`)

Thoughts?

**Comment by ambrozic** [comment]
Based on what we discussed option 3 sound like a reasonable solution. Maybe using [secondary button](https://design-system.service.gov.uk/components/button/#secondary-button-combo-example) for `Return without saving` or simply `Cancel`?

**Comment by paulgain** [comment]
> Based on what we discussed option 3 sound like a reasonable solution. Maybe using [secondary button](https://design-system.service.gov.uk/components/button/#secondary-button-combo-example) for `Return without saving` or simply `Cancel`?

Option 3 it is 👍

**Pull request merged**